test/spec/userSync_spec.js

```diff
@@ -0,0 +1,179 @@
+    getUniqueIdentifierStrStub.restore();
+    timeoutStub.restore();
+  });
+
+  it('should register and fire a pixel URL', () => {
+    userSync = newUserSync();
+    userSync.registerSync('image', 'testBidder', 'http://example.com');
+    userSync.syncUsers();
+    expect(triggerPixelStub.getCall(0)).to.not.be.null;
+    expect(triggerPixelStub.getCall(0).args[0]).to.exist.and.to.equal('http://example.com');
+  });
+
+  it('should clear queue after sync', () => {
+    userSync = newUserSync();
+    userSync.syncUsers();
+    expect(triggerPixelStub.callCount).to.equal(0);
+  });
+
+  it('should delay firing a pixel by the expected amount', () => {
+    userSync = newUserSync();
+    userSync.registerSync('image', 'testBidder', 'http://example.com');
+    // This implicitly tests cookie and browser support
+    userSync.syncUsers(999);
+    expect(timeoutStub.getCall(0).args[1]).to.equal(999);
+  });
+
+  it('should register and fires multiple pixel URLs', () => {
+    userSync = newUserSync();
+    userSync.registerSync('image', 'testBidder', 'http://example.com/1');
+    userSync.registerSync('image', 'testBidder', 'http://example.com/2');
+    userSync.syncUsers();
+    expect(triggerPixelStub.getCall(0)).to.not.be.null;
+    expect(triggerPixelStub.getCall(0).args[0]).to.exist.and.to.include('http://example.com/');
+    expect(triggerPixelStub.getCall(1)).to.not.be.null;
+    expect(triggerPixelStub.getCall(1).args[0]).to.exist.and.to.include('http://example.com/');
+    expect(triggerPixelStub.getCall(2)).to.be.null;
+  });
+
+  it('should not register pixel URL since it is not supported', () => {
+    userSync = newUserSync(getUserSyncConfig({pixelEnabled: false}));
+    userSync.registerSync('image', 'testBidder', 'http://example.com');
+    userSync.syncUsers();
+    expect(triggerPixelStub.getCall(0)).to.be.null;
+  });
+
+  it('should register and load an iframe', () => {
+    userSync = newUserSync(getUserSyncConfig({iframeEnabled: true}));
+    userSync.registerSync('iframe', 'testBidder', 'http://example.com/iframe');
+    userSync.syncUsers();
+    let iframe = window.document.getElementById(idPrefix + lastId);
+    expect(iframe).to.exist;
+    expect(iframe.src).to.equal('http://example.com/iframe');
+  });
+
+  it('should only trigger syncs once per page', () => {
+    userSync = newUserSync(getUserSyncConfig({pixelEnabled: true}));
+    userSync.registerSync('image', 'testBidder', 'http://example.com/1');
+    userSync.syncUsers();
+    userSync.registerSync('image', 'testBidder', 'http://example.com/2');
+    userSync.syncUsers();
+    expect(triggerPixelStub.getCall(0)).to.not.be.null;
+    expect(triggerPixelStub.getCall(0).args[0]).to.exist.and.to.equal('http://example.com/1');
+    expect(triggerPixelStub.getCall(1)).to.be.null;
+  });
+
+  // Since cookie support is only checked when the module is loaded this test will not work, but a test that covers
+  // this scenario is important and this should be revisited.
```

Comment:
If you make the tweaks to `userSync` (see the comment in that file), this becomes testable very easily.

```
userSync = newUserSync({
  config: ...,
  cookiesAreSupported: false
})
```